src/Perl6/Metamodel/ClassHOW.nqp

```diff
@@ -54,7 +55,9 @@ class Perl6::Metamodel::ClassHOW
             $new_type := nqp::newtype($metaclass, $repr);
         }
         my $obj := nqp::settypehll($new_type, 'Raku');
-        $metaclass.set_name($obj, $name // "<anon|{$anon_id++}>");
+        $id_lock.protect: {
+            $metaclass.set_name($obj, $name // "<anon|{$anon_id++}>");
+        }
```

Comment:
Probably won't make a difference, but you could improve parallelization by only protecting `$anon_id++` like in https://github.com/rakudo/rakudo/blob/22f78f4e6c87dcff25b5ce971b1d354a6cf0fc1f/src/Perl6/Grammar.nqp#L484

Reply:
I thought about it, but how much sense does it make? The current way is more readable.

In either case, `nqp::cas_i` would be even better here, but NQP seemingly doesn't support it.